AndroidX support required

# Issue
This library does not work with AndroidX since it uses the support widget for RecyclerView

```
/Users/mofa/Projects/android/app/build/generated/source/kapt/stagingFakebrandDebug/com/allco/ui/bottomsheet/databinding/BottomSheetListBinding.java:21: error: package android.support.v7.widget does not exist
    public final android.support.v7.widget.RecyclerView rvBottomSheet;
                                          ^
/Users/mofa/Projects/android/app/build/generated/source/kapt/stagingFakebrandDebug/com/allco/ui/bottomsheet/databinding/BottomSheetListBinding.java:34: error: package android.support.v7.widget does not exist
        this.rvBottomSheet = (android.support.v7.widget.RecyclerView) bindings[0];
                                                       ^
Note: /Users/mofa/Projects/android/app/build/generated/source/kapt/stagingFakebrandDebug/com/drivekitt/mcar/databinding/ComponentChippyBindingImpl.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
```

--- a/bottomsheet/build.gradle
+++ b/bottomsheet/build.gradle
@@ -12,7 +12,7 @@ android {
     defaultConfig {
         minSdkVersion 19
         targetSdkVersion 28
-        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
     }
 
     buildTypes {
@@ -70,12 +70,12 @@ dependencies {
     ktlint "com.github.shyiko:ktlint:0.28.0"
     implementation fileTree(dir: 'libs', include: ['*.jar'])
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
-    implementation 'com.android.support:appcompat-v7:28.0.0-rc02'
-    implementation "com.android.support:design:28.0.0-rc02"
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'com.google.android.material:material:1.0.0'
     testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
-    implementation 'com.android.support:recyclerview-v7:28.0.0-rc02'
+    androidTestImplementation 'androidx.test:runner:1.1.0'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.0'
+    implementation 'androidx.recyclerview:recyclerview:1.0.0'
     implementation 'com.squareup.picasso:picasso:2.5.2'
 }
 repositories {

--- a/bottomsheet/build.gradle
+++ b/bottomsheet/build.gradle
@@ -8,7 +8,6 @@ group = 'com.github.allco'
 
 android {
     compileSdkVersion 28
-    buildToolsVersion "28.0.2"
 
     defaultConfig {
         minSdkVersion 19

--- a/bottomsheet/src/main/java/com/allco/ui/bottomsheet/BottomSheetBuilder.kt
+++ b/bottomsheet/src/main/java/com/allco/ui/bottomsheet/BottomSheetBuilder.kt
@@ -3,13 +3,13 @@ package com.allco.ui.bottomsheet
 import android.app.Activity
 import android.content.Context
 import android.content.DialogInterface
-import android.databinding.ViewDataBinding
+import androidx.databinding.ViewDataBinding
 import android.graphics.drawable.Drawable
-import android.support.annotation.ColorRes
-import android.support.annotation.DrawableRes
-import android.support.annotation.StringRes
-import android.support.annotation.StyleRes
-import android.support.v4.app.Fragment
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.annotation.StyleRes
+import androidx.fragment.app.Fragment
 
 /**
  * Creates [BottomSheetBuilder]

--- a/bottomsheet/src/main/java/com/allco/ui/bottomsheet/BottomSheetDialog.kt
+++ b/bottomsheet/src/main/java/com/allco/ui/bottomsheet/BottomSheetDialog.kt
@@ -6,17 +6,18 @@ import android.graphics.Color
 import android.graphics.drawable.ColorDrawable
 import android.os.Build
 import android.os.Bundle
-import android.support.design.R
-import android.support.design.widget.BottomSheetBehavior
+import com.google.android.material.R
+import com.google.android.material.bottomsheet.BottomSheetBehavior
 import android.util.TypedValue
 import android.view.WindowManager
 import android.widget.FrameLayout
 import com.allco.ui.bottomsheet.databinding.BottomSheetListBinding
 import com.allco.ui.recyclerView.ObserverBasedAdapter
+import com.google.android.material.bottomsheet.BottomSheetDialog
 import kotlin.math.min
 import kotlin.math.roundToInt
 
-class BottomSheetDialog(context: Context) : android.support.design.widget.BottomSheetDialog(context) {
+class BottomSheetDialog(context: Context) : BottomSheetDialog(context) {
 
     internal fun init(settings: BottomSheetSettings) {
         val binding = BottomSheetListBinding.inflate(layoutInflater)

--- a/bottomsheet/src/main/java/com/allco/ui/bottomsheet/Extensions.kt
+++ b/bottomsheet/src/main/java/com/allco/ui/bottomsheet/Extensions.kt
@@ -2,8 +2,8 @@ package com.allco.ui.bottomsheet
 
 import android.app.Dialog
 import android.graphics.drawable.Drawable
-import android.support.annotation.DrawableRes
-import android.support.v7.content.res.AppCompatResources
+import androidx.annotation.DrawableRes
+import androidx.appcompat.content.res.AppCompatResources
 
 fun Dialog.getDrawable(@DrawableRes drawableRes: Int?): Drawable? =
     drawableRes?.let { AppCompatResources.getDrawable(context, drawableRes) }

--- a/bottomsheet/src/main/java/com/allco/ui/bottomsheet/ViewModels.kt
+++ b/bottomsheet/src/main/java/com/allco/ui/bottomsheet/ViewModels.kt
@@ -1,7 +1,7 @@
 package com.allco.ui.bottomsheet
 
 import android.content.DialogInterface
-import android.databinding.ViewDataBinding
+import androidx.databinding.ViewDataBinding
 import android.graphics.drawable.Drawable
 import com.allco.ui.recyclerView.ObserverBasedAdapter
 

--- a/bottomsheet/src/main/java/com/allco/ui/recyclerView/BindingAdapters.kt
+++ b/bottomsheet/src/main/java/com/allco/ui/recyclerView/BindingAdapters.kt
@@ -1,23 +1,24 @@
 package com.allco.ui.recyclerView
 
-import android.databinding.BindingAdapter
+import androidx.databinding.BindingAdapter
 import android.graphics.Bitmap
 import android.graphics.Color
 import android.graphics.drawable.BitmapDrawable
 import android.graphics.drawable.ColorDrawable
 import android.graphics.drawable.Drawable
-import android.support.annotation.ColorRes
-import android.support.annotation.DrawableRes
-import android.support.graphics.drawable.VectorDrawableCompat
-import android.support.v4.content.ContextCompat
-import android.support.v4.graphics.drawable.DrawableCompat
-import android.support.v7.widget.RecyclerView
+import androidx.annotation.ColorRes
+import androidx.annotation.DrawableRes
+import androidx.vectordrawable.graphics.drawable.VectorDrawableCompat
+import androidx.core.content.ContextCompat
+import androidx.core.graphics.drawable.DrawableCompat
+import androidx.recyclerview.widget.RecyclerView
 import android.text.TextUtils.isEmpty
 import android.util.Log
 import android.view.View
 import android.view.ViewGroup
 import android.widget.TextView
 import com.squareup.picasso.Picasso
+import com.squareup.picasso.Target
 
 @BindingAdapter(value = ["listItems"])
 fun setRecyclerViewItems(recyclerView: RecyclerView, listItems: ObserverBasedAdapter.ItemList) {
@@ -102,7 +103,7 @@ fun loadImageUrlDrawableLeft(textView: TextView, imageUrlDrawableLeft: String?, 
     }
 }
 
-private class TextViewPicassoTarget internal constructor(private val textView: TextView) : com.squareup.picasso.Target {
+private class TextViewPicassoTarget internal constructor(private val textView: TextView) : Target {
     override fun onBitmapLoaded(bitmap: Bitmap, from: Picasso.LoadedFrom) {
         val drawable = BitmapDrawable(textView.resources, bitmap)
         drawable.setBounds(0, 0, bitmap.width, bitmap.height)

--- a/bottomsheet/src/main/java/com/allco/ui/recyclerView/ObserverBasedAdapter.kt
+++ b/bottomsheet/src/main/java/com/allco/ui/recyclerView/ObserverBasedAdapter.kt
@@ -1,12 +1,13 @@
 package com.allco.ui.recyclerView
 
-import android.databinding.DataBindingUtil
-import android.databinding.ObservableArrayList
-import android.databinding.ViewDataBinding
-import android.support.annotation.LayoutRes
-import android.support.v7.widget.RecyclerView
+import androidx.databinding.DataBindingUtil
+import androidx.databinding.ObservableArrayList
+import androidx.databinding.ViewDataBinding
+import androidx.annotation.LayoutRes
+import androidx.recyclerview.widget.RecyclerView
 import android.view.LayoutInflater
 import android.view.ViewGroup
+import androidx.databinding.ObservableList
 import com.allco.ui.bottomsheet.BR
 
 open class ObserverBasedAdapter(private val data: ItemList) : RecyclerView.Adapter<ObserverBasedAdapter.ViewHolder>() {
@@ -23,7 +24,7 @@ open class ObserverBasedAdapter(private val data: ItemList) : RecyclerView.Adapt
     }
 
     init {
-        data.addOnListChangedCallback(object : android.databinding.ObservableList.OnListChangedCallback<ItemList>() {
+        data.addOnListChangedCallback(object : ObservableList.OnListChangedCallback<ItemList>() {
             override fun onChanged(sender: ItemList?) = notifyDataSetChanged()
             override fun onItemRangeRemoved(sender: ItemList?, positionStart: Int, itemCount: Int) = notifyItemRangeRemoved(positionStart, itemCount)
             override fun onItemRangeMoved(sender: ItemList?, fromPosition: Int, toPosition: Int, itemCount: Int) = notifyDataSetChanged()

--- a/bottomsheet/src/main/res/layout/bottom_sheet_list.xml
+++ b/bottomsheet/src/main/res/layout/bottom_sheet_list.xml
@@ -14,7 +14,7 @@
             />
     </data>
 
-    <android.support.v7.widget.RecyclerView
+    <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/rv_bottom_sheet"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
@@ -23,7 +23,7 @@
         android:clickable="false"
         android:paddingBottom="@dimen/bottom_sheet_padding_bottom"
         android:paddingTop="@dimen/bottom_sheet_padding_bottom"
-        bind:layoutManager="android.support.v7.widget.LinearLayoutManager"
+        bind:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
         bind:listItems="@{items}" />
 
 </layout>

--- a/bottomsheet/src/main/res/layout/bottom_sheet_list_item.xml
+++ b/bottomsheet/src/main/res/layout/bottom_sheet_list_item.xml
@@ -16,7 +16,7 @@
         android:id="@+id/bottom_sheet_item_title"
         android:layout_width="match_parent"
         android:layout_height="@dimen/bottom_sheet_text_height"
-        android:background="?android:attr/selectableItemBackground"
+        android:background="?attr/selectableItemBackground"
         android:clickable="true"
         android:contentDescription="@{model.title}"
         android:drawablePadding="@dimen/bottom_sheet_padding_double"

--- a/build.gradle
+++ b/build.gradle
@@ -1,7 +1,7 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-    ext.kotlin_version = '1.2.70'
+    ext.kotlin_version = '1.3.10'
     ext.anko_version = '0.10.6'
 
     repositories {
@@ -9,7 +9,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.1.4'
+        classpath 'com.android.tools.build:gradle:3.2.1'
         classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
         classpath 'com.github.dcendents:android-maven-gradle-plugin:2.0'
         classpath "com.github.ben-manes:gradle-versions-plugin:0.20.0"

--- a/example/build.gradle
+++ b/example/build.gradle
@@ -13,7 +13,7 @@ android {
         versionCode 1
         versionName "1.0"
 
-        testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
 
     }
 
@@ -67,14 +67,14 @@ dependencies {
     implementation project(':bottomsheet')
     implementation fileTree(dir: 'libs', include: ['*.jar'])
 
-    implementation 'com.android.support:appcompat-v7:28.0.0-rc02'
-    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    implementation 'androidx.appcompat:appcompat:1.0.2'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
     testImplementation 'junit:junit:4.12'
-    androidTestImplementation 'com.android.support.test:runner:1.0.2'
-    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    androidTestImplementation 'androidx.test:runner:1.1.0'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.1.0'
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
 
-    implementation 'com.android.support:recyclerview-v7:28.0.0-rc02'
+    implementation 'androidx.recyclerview:recyclerview:1.0.0'
     // Anko Commons
     implementation "org.jetbrains.anko:anko-commons:$anko_version"
 }

--- a/example/build.gradle
+++ b/example/build.gradle
@@ -5,7 +5,6 @@ apply plugin: "com.github.ben-manes.versions"
 
 android {
     compileSdkVersion 28
-    buildToolsVersion "28.0.2"
 
     defaultConfig {
         applicationId "com.allco.ui.bottomsheet.example"

--- a/example/src/main/java/com/allco/ui/bottomsheet/example/MainActivity.kt
+++ b/example/src/main/java/com/allco/ui/bottomsheet/example/MainActivity.kt
@@ -1,10 +1,10 @@
 package com.allco.ui.bottomsheet.example
 
-import android.databinding.DataBindingUtil
+import androidx.databinding.DataBindingUtil
 import android.graphics.Color
 import android.os.Bundle
-import android.support.v4.content.res.ResourcesCompat
-import android.support.v7.app.AppCompatActivity
+import androidx.core.content.res.ResourcesCompat
+import androidx.appcompat.app.AppCompatActivity
 import android.view.View
 import com.allco.ui.bottomsheet.bottomSheet
 import com.allco.ui.bottomsheet.example.databinding.ActivityMainBinding

--- a/example/src/main/res/layout/custom_layout.xml
+++ b/example/src/main/res/layout/custom_layout.xml
@@ -3,7 +3,7 @@
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools">
 
-    <android.support.constraint.ConstraintLayout
+    <androidx.constraintlayout.widget.ConstraintLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         tools:context=".MainActivity">
@@ -42,5 +42,5 @@
             app:layout_constraintStart_toStartOf="parent"
             app:layout_constraintTop_toBottomOf="@+id/textView" />
 
-    </android.support.constraint.ConstraintLayout>
+    </androidx.constraintlayout.widget.ConstraintLayout>
 </layout>

--- a/gradle.properties
+++ b/gradle.properties
@@ -6,6 +6,8 @@
 # http://www.gradle.org/docs/current/userguide/build_environment.html
 # Specifies the JVM arguments used for the daemon process.
 # The setting is particularly useful for tweaking memory settings.
+android.enableJetifier=true
+android.useAndroidX=true
 org.gradle.jvmargs=-Xmx1536m
 # When configured, Gradle will run in incubating parallel mode.
 # This option should only be used with decoupled projects. More details, visit

